### ISSUE TITLE
error establishing a database connection on local server

### ISSUE THREAD

**Issue by kellerchch** [postlight/headless-wp-starter#88]
This is the error I am getting after using nedstark/winteriscoming as username and password. What should my other settings be to get the wordpress back end set up? 

Error establishing a database connection
This either means that the username and password information in your wp-config.php file is incorrect or we can’t contact the database server at localhost. This could mean your host’s database server is down.

Are you sure you have the correct username and password?
Are you sure that you have typed the correct hostname?
Are you sure that the database server is running?

**Comment by tjcafferkey** [comment]
I've literally just encountered this problem myself! Be good to know the solution but trying to have a dig around myself at the same time.

**Comment by tjcafferkey** [comment]
@kellerchch have you managed to solve this yet? Interested in the solution.

**Comment by ginatrapani** [comment]
Is mysql running? If you run `mysql -uroot -p` at the command line, what do you get?

**Comment by tjcafferkey** [comment]
I had to run `mysql.server` start which fixed my problem.

**Comment by kellerchch** [comment]
It does seem that mysql may not have been installed. I manually downloaded and then discovered two versions of myqsl are on my machine now. I started one and stopped the other but the wordpress configuration screen doesn't know about the database and now I see that I need to set up a table first. Not sure the best way to do that on my mac. phpmyadmin software won't download because host servers seem to be down. I'll try again another day. Thanks for replying.

**Comment by glrodasz** [comment]
I had a similar issue after installing another software that probably changed my hosts file configuration. The workaround is to change the `wp-config.php` DB_HOST to use the local ip instead:

```
/** MySQL hostname */
define('DB_HOST', '127.0.0.1');
```

**Comment by trevorlitsey** [comment]
I am having a similar issue. I'm able to log in to mysql via the command line but can't get a connection from wordpress. tried changing the DB_HOST to '127.0.0.1' but that did not work either.

**Comment by glrodasz** [comment]
@trevorlitsey Check my answer here it could be related with your problem: https://github.com/postlight/headless-wp-starter/issues/94#issuecomment-397891514

**Comment by renet** [comment]
`sudo service mysql start` was my solution to make it work.

**Comment by tristanoneil** [comment]
#105 has made some adjustments regarding the MySQL setup. If you're still having issues give it a shot.

**Comment by annamirakk** [comment]
I am having a similar issue. I've just installed the latest version of headless-wp-starter with Docker, then restarter my machine and can't connect to the database. I'm on MacOS Mojave 10.14.2, below are the commands I ran from the project directory:
`mysql.server start`
`docker exec -it wp-headless /bin/bash`
`cd frontend && yarn start`

I see "Compiled successfully" in the console, but an error on the frontend:
"invalid json response body at http://localhost:8080/wp-json/menus/v1/menus/header-menu reason: Unexpected token < in JSON at position 0
FetchError: invalid json response body at http://localhost:8080/wp-json/menus/v1/menus/header-menu reason: Unexpected token < in JSON at position 0
    at /var/www/html/frontend/node_modules/node-fetch/lib/body.js:48:31
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)"

and an "Error establishing a database connection" error in the backend. Any help is much appreciated!
